sources/rest_api/paginators.py

```diff
@@ -133,19 +134,46 @@ class JSONResponsePaginator(BaseNextUrlPaginator):
 
     def __init__(
         self,
-        next_key: Union[str, Sequence[str]] = "next",
+        next_url_path: jsonpath.TJsonPath = "next",
     ):
         """
         Args:
-            next_key (str, optional): The key in the JSON response that
-                contains the next page URL. Defaults to 'next'.
+            next_url_path: The JSON path to the key that contains the next page URL in the response.
+                Defaults to 'next'.
         """
         super().__init__()
-        self.next_key = next_key
-        self._next_key_accessor = create_nested_accessor(next_key)
+        self.next_url_path = jsonpath.compile_path(next_url_path)
```

Comment:
@steinitzu thanks for the suggestion. Could you please elaborate on the real use case for jsonpath here? I can imagine that there might be a non-trivial path required to extract data records (e.g., nested dictionaries plus list indices). Is this the case for the data at the next url?

Reply:
I'm thinking mostly about consistency, since we use json path in similar places, like the data path, incremental cursor, etc.  
I don't think otherwise having this buried in lists and such is common, but can be useful for re-using pagination config between endpoints when the API is not consistent. E.g. if the key is sometimes top level, sometimes wrapped in an envelope. Jsonpath `"$|*.next"` would work for all endpointds in that case.